[Koh Chen Huang, Samuel] iP

Samsation opened the pull request:
# dukeBot Ver. Pingu
>"A computer once beat me at chess, but it was no match for me at kick boxing." – Emo Philips
([source](https://www.brainyquote.com/quotes/emo_philips_128993))

dukeBot Ver. Pingu frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ _SUPER FAST_ to use

All you need to do is, 
1. download it from [here](https://github.com/Samsation/ip)
2. double-click it
3. add your tasks
4. let it manage your tasks for you :wink:

And it is **FREE**!

## Features
- [x] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public static void main(String[] args) {
        Application.launch(Main.class, args);
    }

Commit:
Add Gradle support

Samsation committed:
Implement Level-1

Samsation committed:
Implement Level-2

Samsation committed:
Fix list counter

Samsation committed:
Implement Level-3

Samsation committed:
Implement Level-4

Samsation committed:
Implement Automated Text UI Testing

Samsation committed:
Implement Level-5

Samsation committed:
Implement Level-6

Samsation committed:
Implement Enums

Samsation committed:
Update test cases for Automated Text UI Testing

Samsation committed:
Implement Level-7

Samsation committed:
Fix DukeException

Samsation committed:
Revert "Fix DukeException"

This reverts commit b255c95b4d42717f0909a3ff13157a11aa03242f.

Samsation committed:
Fix OOP

Samsation committed:
Fix original classes

Samsation committed:
Implement Level-8

Samsation committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/DeadlineCommand.java
#	src/main/java/Event.java
#	src/main/java/EventCommand.java
#	src/main/java/Parser.java
#	src/main/java/Task.java
#	src/main/java/TaskList.java

Samsation committed:
Implement Use More OOP and Organize into Packages

Samsation committed:
Merge branch 'add-gradle-support' of https://github.com/Samsation/ip

Samsation committed:
Add Gradle support

Samsation committed:
Add JUnit Tests

Samsation committed:
Implement Level-9

Samsation committed:
Improve UI

Samsation committed:
Add JavaDoc comments to duke.command package

Samsation committed:
Add JavaDoc comments to duke.task package

Samsation committed:
Add JavaDoc comments to duke package

Samsation committed:
Fix code to comply with coding standard

Samsation committed:
Merge branch 'branch-A-CodingStandard'

Samsation committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/Ui.java
#	src/main/java/duke/command/ByeCommand.java
#	src/main/java/duke/command/Command.java
#	src/main/java/duke/command/DeadlineCommand.java
#	src/main/java/duke/command/DeleteCommand.java
#	src/main/java/duke/command/EventCommand.java
#	src/main/java/duke/command/ListCommand.java
#	src/main/java/duke/command/MarkCommand.java
#	src/main/java/duke/command/ToDoCommand.java
#	src/main/java/duke/command/UnmarkCommand.java
#	src/main/java/duke/task/Event.java
#	src/main/java/duke/task/Task.java
#	src/main/java/duke/task/ToDo.java

Samsation committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/Ui.java
#	src/main/java/duke/command/DeleteCommand.java
#	src/main/java/duke/task/Task.java

Samsation committed:
Add more JavaDoc comments

Samsation committed:
Use CheckStyle to detect coding style violations

Samsation committed:
Use GFMD

Samsation committed:
Implement Level-10

Samsation committed:
Revert "Implement Level-10"

This reverts commit 26d9d1db7e69078a804a14d81620c216430a3120.

Samsation committed:
Add JavaDoc comments for GUI

Samsation committed:
Merge branch 'branch-Level-10'

Samsation committed:
Run CheckStyle without JavaDoc comments checks

Samsation committed:
Run CheckStyle with JavaDoc comments checks

Samsation committed:
Merge branch 'branch-A-CheckStyle'

Samsation committed:
Fix Branching

Samsation committed:
Add Java Varargs

Samsation committed:
Merge branch 'branch-A-Varargs'

Samsation committed:
Add JavaDoc comments

Samsation committed:
Update README.md

lkwlkww commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,84 @@
+    }
+
+    /**
+     * A method that flips the dialog box such that the ImageView is on the left and text on the right.
```
Perhaps you could change your method header to start with 'Flips'? According to the provided [coding standards](https://se-education.org/guides/conventions/java/intermediate.html), method headers should start with words like 'Returns', 'Sends', 'Adds' etc

Samsation replied:
Fixed under A-CodeQuality. Thank you!

lkwlkww commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,106 @@
+    public static Command parse(String inputCommand) throws DukeException {
+        try {
+            String[] splitInputCommand = inputCommand.trim().split(" ", 2);
+            CommandTag ct = CommandTag.valueOf(splitInputCommand[0].toUpperCase());
```
According to the coding standards:
> scratch variables used for temporary storage or indices can be kept short

I'm not sure if `ct` here fits the criteria, or if it should be changed to something more meaningful like `commandTag`. What do you think?

lkwlkww commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,84 @@
+    }
+
+    /**
+     * A method that retrieves the user's dialog.
```
Perhaps you could consider not including method headers for getters in future, if you would like to save some effort:
> You MUST write header comments for all classes, public methods. But they can be omitted for the following cases:
i. Getters/setters

lkwlkww commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,52 @@
+     * @param stage The current Stage of the Application.
+     */
+    public Duke(Stage stage) {
+        this.ui = new Ui(stage);
```
I think you could consider leaving the `this` keyword out, according to the following coding standard rule:
> Use the this keyword only when a field is shadowed by a method or constructor parameter.

lkwlkww reviewed (commented):
Very well-adhered to the coding standards! It was hard to find deviations.

ThomasHoooo commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,84 @@
+     * A method that flips the dialog box such that the ImageView is on the left and text on the right.
+     */
+    private void flip() {
+        ObservableList<Node> tmp = FXCollections.observableArrayList(this.getChildren());
```
Maybe you can set a more meaningful name instead of just tmp?

Samsation replied:
Fixed under A-CodeQuality. Thank you!

ThomasHoooo reviewed (commented):
Inconsistent styling of class description.

Samsation committed:
Update JavaDoc comments

Samsation committed:
Add assertions to parse() method

This assertion ensures that the user input commands will never result
in an empty array when splitting the input command.

Samsation committed:
Improve code quality

There are several chunks of code clustered together.

Cleaning up JavaDoc comments and variable naming allows for better
readability. Increasing usage of whitespaces helps in this regard as
well.

Samsation committed:
Merge pull request #1 from Samsation/branch-A-Assertions

Add assertions to parse() method

Samsation committed:
Merge branch 'master' into branch-A-CodeQuality

Samsation committed:
Merge pull request #2 from Samsation/branch-A-CodeQuality

branch-A-CodeQuality

Samsation committed:
Use streams for displaying of TaskList

Iterating through the TaskList requires multiple loops to properly
format the string output.

Let's use a sequence of stream operations to perform string
concatenation for the Tasks in the TaskList.

Using stream operations is preferable as it condenses the code and
increases readability.

Samsation committed:
Merge pull request #3 from Samsation/branch-A-Streams

branch-A-Streams

Samsation committed:
Set up Continuous Integration (CI)

Samsation committed:
Add Update command

Samsation committed:
Fix JUnit tests

Samsation committed:
Improve GUI

Samsation committed:
Improve JavaDoc comments

Samsation committed:
Add user guide and product screenshot

Samsation committed:
Update JavaFX version
